clippy_lints/src/methods/iter_any.rs

```diff
@@ -0,0 +1,60 @@
+use clippy_utils::diagnostics::span_lint;
+use rustc_hir::{Expr, ExprKind};
+use rustc_lint::LateContext;
+use rustc_middle::ty::{self};
+
+use super::{SLICE_ITER_ANY, UNNECESSARY_ITER_ANY, method_call};
+
+pub(super) fn check(cx: &LateContext<'_>, expr: &Expr<'_>) {
+    if !expr.span.from_expansion()
+    // any()
+    && let Some((name, recv, args, _, _)) = method_call(expr)
+    && name == "any"
+    // check if the inner closure is a equality check
+    && args.len() == 1
+    && let ExprKind::Closure(closure) = args[0].kind
+    && let body = cx.tcx.hir().body(closure.body)
+    && let ExprKind::Binary(op, _, _) = body.value.kind
+    && op.node == rustc_ast::ast::BinOpKind::Eq
+    // iter()
+    && let Some((name, recv, _, _, _)) = method_call(recv)
+    && name == "iter"
+    {
+        let ref_type = cx.typeck_results().expr_ty(recv);
```

Comment:
issue: Probably should be [`expr_ty_adjusted`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/typeck_results/struct.TypeckResults.html#method.expr_ty_adjusted) to handle autoderef. Add a test that ensures this works on a vector `vec.iter().any(...)`.

Reply:
Thank you! I changed to use `expr_ty_adjusted` and added a test for this in https://github.com/rust-lang/rust-clippy/pull/13817/commits/e964cbc8380547e13ca85cd8563de796fb7b311d.